library/core/src/clone.rs

```diff
@@ -38,7 +38,25 @@
 
 mod uninit;
 
-/// A common trait for the ability to explicitly duplicate an object.
+/// A common trait that allows explicit creation of a "duplicate" value.
+///
+/// ## Important Note on "Duplication"
+///
+/// What "duplication" means depends on the type implementing `Clone`:
+///
+/// - For most types, calling [`clone`] creates a semantically independent copy with its own
+///   unique identity and state.
+/// - For smart pointers (like [`Arc`], [`Rc`]), [`clone`] increases the reference count but points
+///   to the same underlying data. This means modifications to the underlying data through one
+///   clone will be visible through other clones.
+/// - For reference types (`&T`), [`clone`] just creates another reference to the same value.
+///
+/// This distinction is especially important when using `#[derive(Clone)]` on structs containing
+/// smart pointers like `Arc<Mutex<T>>` - the cloned struct will share mutable state with the
+/// original.
+///
+/// [`Arc`]: ../../std/sync/struct.Arc.html
+/// [`Rc`]: ../../std/rc/struct.Rc.html
```

Comment:
After this we start talking about how this differs from Copy, you see, and that probably doesn't fit under the same heading.